### DISCUSSION TITLE
Docker image not starting

### DISCUSSION THREAD

**Discussion by AbhijeetShk** [stackernews/stacker.news#916] [local dev help]
**Description**
Hi, I followed the readme process to start the project locally but after running docker compose it doesn't starts the server and keeps giving following output in the terminal 

Waiting for OpenSearch to start...
worker        | (node:30) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 removeListener listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
worker        |     at __node_internal_genericNodeError (node:internal/errors:865:15)
worker        |     at _addListener (node:events:588:17)
worker        |     at EventEmitter.addListener (node:events:606:10)
worker        |     at module.exports (/app/node_modules/lightning/grpc/handle_remove_listener.js:13:11)
worker        |     at <anonymous> (/app/node_modules/lightning/lnd_methods/invoices/subscribe_to_invoices.js:101:5)
worker        |     at check (/app/node_modules/async/doWhilst.js:61:9)
worker        |     at <anonymous> (/app/node_modules/async/doUntil.js:43:40)
worker        |     at <anonymous> (/app/node_modules/lightning/lnd_methods/invoices/subscribe_to_invoices.js:172:12)
worker        |     at <anonymous> (/app/node_modules/async/doUntil.js:43:9)
worker        |     at next (/app/node_modules/async/doWhilst.js:54:9)
worker        |     at Timeout._onTimeout (/app/node_modules/lightning/lnd_methods/invoices/subscribe_to_invoices.js:125:16)
worker        |     at listOnTimeout (node:internal/timers:569:17)
worker        |     at process.processTimers (node:internal/timers:512:7)
os-dashboard  | {"type":"log","@timestamp":"2024-03-10T09:01:29Z","tags":["error","opensearch","data"],"pid":1,"message":"[ConnectionError]: connect ECONNREFUSED 172.18.0.4:9200"}
worker        | [
worker        |   503,
worker        |   'UnexpectedInvoiceSubscriptionError',
worker        |   {
worker        |     err: Error: 14 UNAVAILABLE: Failed to parse DNS address dns:<YOUR LND GRPC HOST>:<YOUR LND GRPC PORT>
worker        |         at callErrorFromStatus (/app/node_modules/@grpc/grpc-js/build/src/call.js:31:19)
worker        |         at Object.onReceiveStatus (/app/node_modules/@grpc/grpc-js/build/src/client.js:357:53)
worker        |         at Object.onReceiveStatus (/app/node_modules/@grpc/grpc-js/build/src/client-interceptors.js:323:181)
worker        |         at <anonymous> (/app/node_modules/@grpc/grpc-js/build/src/resolving-call.js:99:78)
worker        |         at process.processTicksAndRejections (node:internal/process/task_queues:77:11)
worker        |     for call at
worker        |         at ServiceClientImpl.makeServerStreamRequest (/app/node_modules/@grpc/grpc-js/build/src/client.js:340:32)
worker        |         at ServiceClientImpl.<anonymous> (/app/node_modules/@grpc/grpc-js/build/src/make-client.js:105:19)
worker        |         at <anonymous> (/app/node_modules/lightning/lnd_methods/invoices/subscribe_to_invoices.js:95:43)
worker        |         at check (/app/node_modules/async/doWhilst.js:61:9)
worker        |         at doWhilst (/app/node_modules/async/doWhilst.js:64:12)
worker        |         at awaitable (/app/node_modules/async/internal/awaitify.js:14:28)
worker        |         at doUntil (/app/node_modules/async/doUntil.js:41:27)
worker        |         at module.exports (/app/node_modules/lightning/lnd_methods/invoices/subscribe_to_invoices.js:90:3)
worker        |         at file:///app/worker/wallet.js:57:17 {
worker        |       code: 14,
worker        |       details: 'Failed to parse DNS address dns:<YOUR LND GRPC HOST>:<YOUR LND GRPC PORT>',
worker        |       metadata: [Metadata]
worker        |     }
worker        |   }
worker        | ]
worker        | error subscribing - trying again
worker        | [
worker        |   503,
worker        |   'UnexpectedPaymentsSubErr',
worker        |   {
worker        |     err: Error: 14 UNAVAILABLE: Failed to parse DNS address dns:<YOUR LND GRPC HOST>:<YOUR LND GRPC PORT>
worker        |         at callErrorFromStatus (/app/node_modules/@grpc/grpc-js/build/src/call.js:31:19)
worker        |         at Object.onReceiveStatus (/app/node_modules/@grpc/grpc-js/build/src/client.js:357:53)
worker        |         at Object.onReceiveStatus (/app/node_modules/@grpc/grpc-js/build/src/client-interceptors.js:323:181)
worker        |         at <anonymous> (/app/node_modules/@grpc/grpc-js/build/src/resolving-call.js:99:78)
worker        |         at process.processTicksAndRejections (node:internal/process/task_queues:77:11)
worker        |     for call at
worker        |         at ServiceClientImpl.makeServerStreamRequest (/app/node_modules/@grpc/grpc-js/build/src/client.js:340:32)
worker        |         at ServiceClientImpl.<anonymous> (/app/node_modules/@grpc/grpc-js/build/src/make-client.js:105:19)
worker        |         at module.exports (/app/node_modules/lightning/lnd_methods/offchain/subscribe_to_payments.js:109:31)
worker        |         at file:///app/worker/wallet.js:182:17
worker        |         at file:///app/worker/wallet.js:23:21
worker        |         at new Promise (<anonymous>)
worker        |         at retry.forever (file:///app/worker/wallet.js:22:20)
worker        |         at RetryOperation.runAttempt (/app/node_modules/async-retry/lib/index.js:42:15)
worker        |         at Timeout._onTimeout (/app/node_modules/retry/lib/retry_operation.js:85:10)
worker        |         at listOnTimeout (node:internal/timers:569:17)
worker        |         at process.processTimers (node:internal/timers:512:7) {
worker        |       code: 14,
worker        |       details: 'Failed to parse DNS address dns:<YOUR LND GRPC HOST>:<YOUR LND GRPC PORT>',
worker        |       metadata: [Metadata]
worker        |     }
worker        |   }
worker        | ]
worker        | error subscribing - trying again
opensearch    | curl: (7) Failed to connect to localhost port 9200 after 0 ms: Couldn't connect to server
opensearch    | Waiting for OpenSearch to start...
opensearch    | curl: (7) Failed to connect to localhost port 9200 after 0 ms: Couldn't connect to server
opensearch    | Waiting for OpenSearch to start...
os-dashboard  | {"type":"log","@timestamp":"2024-03-10T09:01:32Z","tags":["error","opensearch","data"],"pid":1,"message":"[ConnectionError]: connect ECONNREFUSED 172.18.0.4:9200"}

**Steps to Reproduce**
Clone and run docker-compose up --build

**Additional context**
I only needed to start this project locally and I guess problem is not having env vars but why do I need them when I just need to run project locally for contributing

**Comment by huumn** [comment]
I'm rewriting the docker setup in this branch: https://github.com/stackernews/stacker.news/pull/900

It should be stable enough to try, better than the other setup, but its probably not free of issues yet.
